packages/@react-aria/button/src/useButton.ts

```diff
@@ -93,18 +93,22 @@ export function useButton(props: AriaButtonProps, ref: RefObject<HTMLElement>):
   let {focusableProps} = useFocusable(props, ref);
   let handlers = mergeProps(pressProps, focusableProps);
   let interactions = mergeProps(contextProps, handlers);
+  let ariaPressed = isSelected ? true : undefined;
 
   return {
     isPressed, // Used to indicate press state for visual
     buttonProps: mergeProps(interactions, {
       id,
       'aria-label': ariaLabel,
       'aria-labelledby': ariaLabelledby,
-      'aria-describedby': ariaDescribedBy,  
+      'aria-describedby': ariaDescribedBy,
       'aria-haspopup': ariaHasPopup,
       'aria-expanded': ariaExpanded || (ariaHasPopup && isSelected),
       'aria-controls': ariaControls,
-      'aria-checked': isSelected,
+      // if an explicit aria-pressed false has been passed int, then use that but as undefined
+      // otherwise, decide for ourselves
+      'aria-pressed': props['aria-pressed'] === false ? undefined : ariaPressed,
+      'aria-checked': props['aria-pressed'] === false ? isSelected : undefined,
```

Comment:
I don't think this is quite right. `aria-pressed` should be `undefined` if it is not a toggle button, and false if it *is* a toggle button but not pressed. Currently you have undefined if it's false as well. See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Toggle_buttons.